docs/REST/authentication.md

```diff
@@ -289,3 +289,74 @@ static class HttpContentExtensions
     }
 }
 ```
+### Java
+```java
+public class ConfigHttpClient {
+
+    public CloseableHttpResponse execute(HttpUriRequest request, Date date, String credential, String secret) {
+        Map<String, String> authHeaders = sign(request, date, credential, secret);
+        authHeaders.forEach(request::setHeader);
+
+        return httpClient.execute(request);
+    }
+
+    private static Map<String, String> sign(HttpUriRequest request, Date date, String credential,
+            String secret) {
+        String requestTime = GMT_DATE_FORMAT.format(date);
+
+        String contentHash = buildContentHash(request);
+
+        // SignedHeaders
+        String signedHeaders = "x-ms-date;host;x-ms-content-sha256";
+
+        // Signature
+        String methodName = request.getRequestLine().getMethod().toUpperCase();
+        URIBuilder uri = new URIBuilder(request.getRequestLine().getUri());
+        String scheme = uri.getScheme() + "://";
+        String requestPath = uri.toString().substring(scheme.length()).substring(uri.getHost().length());
+        String host = new URIBuilder(request.getRequestLine().getUri()).getHost();
+        String toSign = String.format("%s\n%s\n%s;%s;%s", methodName, requestPath, requestTime, host, contentHash);
+
+        // Signature
+        byte[] decodedKey = Base64.getDecoder().decode(secret);
+        String signature = Base64.getEncoder().encodeToString(new HmacUtils(HMAC_SHA_256, decodedKey).hmac(toSign));
+
+        // Compose headers
+        Map<String, String> headers = new HashMap<>();
+        headers.put("x-ms-date", requestTime);
+        headers.put("x-ms-content-sha256", contentHash);
+
+        String authorization = String.format("HMAC-SHA256 Credential=%s, SignedHeaders=%s, Signature=%s",
+                credential, signedHeaders, signature);
+        headers.put("Authorization", authorization);
+
+        headers.put(HttpHeaders.USER_AGENT, USER_AGENT);
+
+        return headers;
+    }
+
+    private static String buildContentHash(HttpUriRequest request) {
+        String content = "";
+        if (request instanceof HttpEntityEnclosingRequest) {
+            try {
+                StringWriter writer = new StringWriter();
+                IOUtils.copy(((HttpEntityEnclosingRequest) request).getEntity().getContent(), writer,
+                        StandardCharsets.UTF_8);
+
+                return writer.toString();
+            }
+            finally {
+                try {
+                    ((HttpEntityEnclosingRequest) request).getEntity().getContent().close();
+                }
+                catch (IOException e) {
+                    LOGGER.trace("Failed to close the input stream.", e);
+                }
+            }
+        }
+
+        byte[] digest = new DigestUtils(SHA_256).digest(content);
```

Comment:
`content` is only initialized as `""` at the beginning. Will this always produces the same result?

Reply:
See your previous comment. line 346 needs to be updated.